packages/synapse-react-client/src/lib/containers/entity_finder/EntityFinderModal.tsx

```diff
@@ -39,39 +38,35 @@ export const EntityFinderModal = (props: EntityFinderModalProps) => {
           setShowUnsavedChangesWarning(false)
         }}
       />
-      <FluidModal
-        className={showUnsavedChangesWarning ? 'SRC-hidden' : undefined}
```

Comment:
Instead of hiding EntityFinderModal when there is an unsaved changes warning, instead show the ConfirmationDialog over the EntityFinderModal -- 

main: 
<img width="1481" alt="main_EntityFinderModal_ConfirmationDialog" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/617a7104-b17b-40f8-ba90-17a0851a41fc">

feature: 
<img width="1457" alt="feature_EntityFinderModal_ConfirmationDialog" src="https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/826f29c8-51b9-4be5-b8e2-39c29b82731f">